lib/miq_expression.rb

```diff
@@ -658,7 +658,13 @@ def self.operands2rubyvalue(operator, ops, context_type)
   end
 
   def self.quote(val, typ)
-    if Field.is_field?(val)
+    valid =
```

Comment:
This method would read better if this logic was moved into a separate method. Perhaps `is_parsable_and_valid?`